read vars from excel sheet

sammarth-k opened the pull request.

sammarth-k committed:
read vars from excel sheet

hdem1 commented on TheRocket/TheRocket.ipynb:
```diff
@@ -10,6 +10,209 @@
+     "output_type": "stream",
+     "text": [
+      "                    Variable  Value Unit  Description  Finalized\n",
+      "0           vehicle_mass_dry     79   lb          NaN        NaN\n",
```
It looks like the code is always displaying NaN for the finalized column. Would it be worthwhile to explicitly set it to False or something in the `readVars` function?

sammarth-k replied:
I fixed this so now it sets that value to the variable and doesn't output the entire table. In the excel sheet the finalized column contains a check box which Aaron had requested, but the only columns the code requires are 'Variable' and 'Value'

hdem1 replied:
Sounds good!

hdem1 commented on TheRocket/TheRocket.ipynb:
```diff
@@ -10,6 +10,209 @@
+   ],
+   "source": [
+    "# STRUCTURES\n",
+    "readVars(\"STRUCTURES2\", 1, 6) # General\n",
```
It looks like "STRUCTURES2" isn't a sheet in the excel file you included in the PR. Should this just be "STRUCTURES"?

sammarth-k replied:
Yup, that's my bad I have fixed that

hdem1 replied:
No worries at all - thanks sm

hdem1 commented on TheRocket/TheRocket.ipynb:
```diff
@@ -26,59 +229,59 @@
-    "vehicle_height = 151 #in\n",
-    "rail_length = 20 #ft\n",
-    "launch_angle = 5 #degrees\n",
+    "# vehicle_mass_dry = 79 #lb (does not include engine)\n",
```
It looks like you are commenting out all of the old declaration which sounds great! Honestly if you have already copied all the code over, you can just delete these lines. It does appear necessary though to remake these variables using the information from your nice excel reading right? Otherwise the cells at the end of the file no longer work if I understand correctly?

sammarth-k replied:
Yep all the lines i commented out are variables inputted in the excel now so it would have no effect whether they are there or not

hdem1 replied:
Sounds good on this front also!

hdem1 commented on TheRocket/TheRocket.ipynb:
```diff
@@ -26,59 +229,59 @@
     "discrete_segments = 100\n",
```
Any reason to leave these general solver parameters as coded values instead of including them in the excel file as well?

sammarth-k replied:
I figured that these are parameters to be tweaked during runtime of the simulations rather than variables for the different groups to input so it might be easier down the road for people working on the flight simulations to tweak these two directly in the notebook

hdem1 replied:
That's fair - thanks for explaining

hdem1 reviewed (changes requested):
Great work on this and fantastic job using git/version control! I included a couple of comments, but would be happy to approve the merge once you respond/address those. Feel free to ping me in the discord with any questions for this merge or future pull requests!

sammarth-k committed:
Edits

1) fixed `readVars` to assign variables rather than printing the table

2) fixed "STRUCTURES2" to "STRUCTURES"

hdem1 commented on TheRocket/TheRocket.ipynb:
```diff
@@ -10,6 +10,163 @@
+    "    # Variable \tValue\tUnit\tDescription\tFinalized\n",
+    "    data.columns = ['Variable', 'Value', 'Unit', 'Description', 'Finalized']\n",
+    "    for i in range(len(data)):\n",
+    "        exec(data['Variable'][i] + ' = ' + str(data['Value'][i]))\n",
```
Very good idea here. I tried running it and am still getting errors when I try to run the cells at the bottom. I think it works if this line is replaced with `exec('global ' + data['Variable'][i] + '; ' + data['Variable'][i] + ' = ' + str(data['Value'][i]))`. Does that work?

sammarth-k replied:
Yeah thats a bit weird, I made the change, everything seems to be good

hdem1 reviewed (changes requested):
Thanks sm for all the changes. Just one more change and I think it should be all good. Feel free to message me again and I can approve it

sammarth-k committed:
Update TheRocket.ipynb

- Fixed global problem
- Program seems to read num_fins as a float so converted to int to avoid bugs
